pkg/infra/scope/cluster_test.go

```diff
@@ -17,14 +17,138 @@ limitations under the License.
+	}{
+		{
+			name: "empty-cni-extra",
+			expect: &Cluster{
+				UID:                 "dcf78fcbb",
+				InfraType:           "aws",
+				NamespacedName:      types.NamespacedName{Namespace: "default", Name: "quickstart"},
+				CredentialSecretRef: "quickstart",
+				Version:             "v1.23.0",
+				Region:              "us-east-1",
+				VpcCIDR:             "10.10.0.0/16",
+				PodCIDR:             []string{"192.168.0.0/16"},
+				ServiceCIDR:         []string{"10.96.0.0/12"},
+				CNIType:             "calico",
+				CNIExtra:            nil,
+				ControlPlane: &Instance{Replicas: 1, InstanceType: "t3.large", SSHKey: "", ImageOS: "",
+					RootVolume: &InstanceVolume{DeviceName: "", Size: 10, Type: "gp3"},
+					DataVolumes: []InstanceVolume{
+						{
+							DeviceName: "/dev/sdb",
+							Size:       10,
+							Type:       "gp2",
+						},
+					},
+				},
+				Workers: []*Instance{
+					{
+						Replicas:     1,
+						InstanceType: "t3.large", SSHKey: "", ImageOS: "",
+						RootVolume: &InstanceVolume{DeviceName: "", Size: 10, Type: "gp2"},
+						DataVolumes: []InstanceVolume{
+							{
+								DeviceName: "/dev/sdb",
+								Size:       10,
+								Type:       "gp3",
+							},
+						}},
+				},
+				EnablePodIdentity: false,
+				BucketName:        "kuratorcluster-dcf78fcbb",
+			},
+		},
+		{
+			name: "cni-extra",
+			expect: &Cluster{
+				UID:                 "dcf78fcbb",
+				InfraType:           "aws",
+				NamespacedName:      types.NamespacedName{Namespace: "default", Name: "quickstart"},
+				CredentialSecretRef: "quickstart",
+				Version:             "v1.23.0",
+				Region:              "us-east-1",
+				VpcCIDR:             "10.10.0.0/16",
+				PodCIDR:             []string{"192.168.0.0/16"},
+				ServiceCIDR:         []string{"10.96.0.0/12"},
+				CNIType:             "calico",
+				CNIExtra: map[string]interface{}{
+					"key1": map[string]interface{}{
+						"key2": "val1",
+					},
+					"key3": "val2",
+				},
+				ControlPlane: &Instance{Replicas: 1, InstanceType: "t3.large", SSHKey: "", ImageOS: "",
+					RootVolume: &InstanceVolume{DeviceName: "", Size: 10, Type: "gp3"},
+					DataVolumes: []InstanceVolume{
+						{
+							DeviceName: "/dev/sdb",
+							Size:       10,
+							Type:       "gp2",
+						},
+					},
+				},
+				Workers: []*Instance{
+					{
+						Replicas:     1,
+						InstanceType: "t3.large", SSHKey: "", ImageOS: "",
+						RootVolume: &InstanceVolume{DeviceName: "", Size: 10, Type: "gp2"},
+						DataVolumes: []InstanceVolume{
+							{
+								DeviceName: "/dev/sdb",
+								Size:       10,
+								Type:       "gp3",
+							},
+						}},
+				},
+				EnablePodIdentity: false,
+				BucketName:        "kuratorcluster-dcf78fcbb",
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			input, err := readFromYaml(tc.name)
+			assert.NoError(t, err)
+
+			tc.expect.Cluster = input
+			assert.Equal(t, tc.expect, NewCluster(input))
+		})
```

Comment:
not sure i understand what is the test coverage

Reply:
make sure `CNIExtra` go the value as expected.